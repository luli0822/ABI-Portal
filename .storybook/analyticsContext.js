export const mockPushCustomLinkEvent = jest.fn(linkName => {});

export const mockUsePageViewEvent = jest.fn();
