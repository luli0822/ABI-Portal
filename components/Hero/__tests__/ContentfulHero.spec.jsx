import React from 'react';
import { shallow } from 'enzyme';
import HomeHero from '../index';
import { mockPushCustomLinkEvent } from '../../../.storybook/analyticsContext';

jest.mock('react', () => {
	const ActualReact = require.requireActual('react');
	return {
		...ActualReact,
		useContext: () => ({
			pushCustomLinkEvent: mockPushCustomLinkEvent,
		}),
	};
});
describe('<HomeHero />', () => {
	it('renders correctly', () => {
		const props = {
			content: {
				title: 'some title',
				body: {},
				primaryLink: {},
				primaryLinkText: 'some text',
				secondaryLink: {},
				secondaryLinkText: 'some text',
			},
			isFirstChild: true,
		};
		// https://github.com/facebook/jest/issues/7802
		// using .debug() instead of adding dependency on enzyme-to-json for now
		// eslint-disable-next-line react/jsx-props-no-spreading
		expect(shallow(<HomeHero {...props} />).debug()).toMatchSnapshot();
	});
});
