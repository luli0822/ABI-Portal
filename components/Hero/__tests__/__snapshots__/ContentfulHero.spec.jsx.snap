// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeHero /> renders correctly 1`] = `
"<section className=\\"contentful-hero\\">
  <div className=\\"contentful-hero__content\\">
    <h1 className=\\"contentful-hero__heading\\">
      some title
    </h1>
    <div className=\\"contentful-hero__detail-row\\">
      <div className=\\"contentful-hero__body-col\\">
        <RichText document={{...}} rootClass=\\"contentful-hero__rich\\" replacements={{...}} />
      </div>
      <div className=\\"contentful-hero__cta-col\\">
        <ContentfulLink linkEntry={{...}} className=\\"contentful-hero__cta\\" onClick={[Function: onClick]} targetLang={[undefined]}>
          some text
        </ContentfulLink>
        <ContentfulLink linkEntry={{...}} className=\\"contentful-hero__cta\\" targetLang={[undefined]}>
          some text
        </ContentfulLink>
      </div>
    </div>
  </div>
</section>"
`;
